### DISCUSSION TITLE
Log received DM

### DISCUSSION THREAD

**Discussion by Steeven-1002** [discordjs/discord.js#9065] [Q&A] [answered]
Hi, I'm trying to (at first) **log a message that a user send to my bot** but I didn't make success... Could someone help me please ? (I'm also French so, sorry for my bad English).

I'm also running on the **14.7.1 version of discord**.

Here is some part of my code:
`
const fs = require('node:fs');
const path = require('node:path');
const { Client, Collection, Events, GatewayIntentBits } = require('discord.js');
const { token } = require('./config.json');
const config = require("./config.json");
Discord = require('discord.js');

const client = new Client({
	intents: ['Guilds', 'GuildMessages', 'MessageContent', 'GuildMembers', 'DirectMessages', 'MessageContent'],
});

client.commands = new Collection();
const commandsPath = path.join(__dirname, 'commands');
const commandFiles = fs.readdirSync(commandsPath).filter(file => file.endsWith('.js'));

for (const file of commandFiles) {
	const filePath = path.join(commandsPath, file);
	const command = require(filePath);
	client.commands.set(command.data.name, command);
}

client.on("message", async message => {
	console.log(`Message recu en DM de ${message.author.username} : ${message.content}`);
	console.log(message);
});`

**Comment by almeidx** [comment, marked as answer]
The `message` event was renamed to `messageCreate` on v14

```diff
- client.on("message", async message => {
+ client.on("messageCreate", async message => {
```

Additionally, you might want to ignore messages that aren't from DMs. For that, you can use `message.channel.isDMBased()`, which returns a boolean

**Reply by Steeven-1002** [reply]
Thank's for your fast answer!

Here is the modified code :

`client.on("messageCreate", async message => { 
	console.log(`Message recu en DM de ${message.author.username} : ${message.content}`);
	console.log(message);
});`

Unfortunately, even with this change, I get nothing in the console...

**Reply by almeidx** [reply]
You're also going to need the `Channel` partial, which you can pass in the `Client` constructor alongside the intents: `partials: [Partials.Channel]`, importing `Partials` from `discord.js`

**Reply by Steeven-1002** [reply]
I'm sorry I'm really bad at coding in javascrpit... How I am suppose to to that? I mean, where I need to write all the things that you say on my code?

**Reply by almeidx** [reply]
Add `Partials` to the discord.js import:
```diff
- const { Client, Collection, Events, GatewayIntentBits } = require('discord.js');
+ const { Client, Collection, Events, GatewayIntentBits, Partials } = require('discord.js');
```

 Pass `partials` in the Client constructor:
```diff
const client = new Client({
  intents: ['Guilds', 'GuildMessages', 'MessageContent', 'GuildMembers', 'DirectMessages', 'MessageContent'],
+ partials: [Partials.Channel],
});
```

**Comment by meesam4687** [comment]
```js
client.on('messageCreate', async message => {
  if (message.channel.isDMBased() && message.author.id !== client.user.id){
       console.log(message.author.username + " : " + messaage.content)
  }
});
```
This should do it

**Reply by jaw0r3k** [reply]
all types are numbers ( in v14 )
so you should use enums like `ChannelType.DM` here
or use function typeguards like mentioned above
